crates/bootstrap_client/src/lib.rs

```diff
@@ -2,24 +2,27 @@
 
 #![deny(missing_docs)]
 
+use base64::Engine;
 use kitsune2_api::{AgentInfoSigned, DynVerifier, K2Error, K2Result, SpaceId};
 use std::sync::Arc;
+use url::Url;
 
 /// Send the agent info, for the given space, to the bootstrap server.
 ///
 /// Note the `blocking_` prefix. This is a hint to the caller that if the function is used in
 /// an async context, it should be treated as a blocking operation.
 pub fn blocking_put(
-    server_url: &str,
+    mut server_url: Url,
     agent_info: &AgentInfoSigned,
 ) -> K2Result<()> {
-    let url = format!(
-        "{server_url}/bootstrap/{}/{}",
```

Comment:
When the `server_url` ends with a `/`, we ended up with `//` here and although that is a valid URL, the bootstrap server was returning a 404... Which I'm pretty sure is wrong, but I'd rather produce simple URLs and not argue with the HTTP server